### PR TITLE
Remove composition-api plugin & upgrade to vue 2.7

### PR DESCRIPTION
**Pull request by marcianos**
See #1978

### CODE REVIEW

File: packages/vue2/vue2-vanilla/config/vue.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /**
  * Switch between Vue 3 and Vue 2 '@vue/composition-api'.
  */
```
Comment (sdirix) [inline packages/vue2/vue2-vanilla/config/vue.ts:3]: We should also adapt the docs

File: packages/vue2/vue2-vanilla/package.json [hunk 2]
```diff
@@ -78,11 +77,11 @@
     "symlink-dir": "^5.0.0",
     "typedoc": "^0.19.2",
     "typescript": "4.2.3",
-    "vue-template-compiler": "^2.6.14"
+    "vue-loader": "^15.10.0"
```
Comment (sdirix) [inline packages/vue2/vue2-vanilla/package.json:80]: Why is `vue-loader` required?

File: packages/vue2/vue2-vanilla/package.json [hunk 3]
```diff
@@ -59,11 +59,10 @@
...
     "@types/jest": "^24.0.23",
-    "@vue/cli-plugin-babel": "~4.5.0",
-    "@vue/cli-plugin-typescript": "~4.5.0",
-    "@vue/cli-plugin-unit-jest": "~4.5.0",
-    "@vue/cli-service": "~4.5.0",
-    "@vue/composition-api": "^1.4.2",
+    "@vue/cli-plugin-babel": "~4.5.18",
+    "@vue/cli-plugin-typescript": "~4.5.18",
+    "@vue/cli-plugin-unit-jest": "~4.5.18",
+    "@vue/cli-service": "~4.5.18",
```
Comment (sdirix) [inline packages/vue2/vue2-vanilla/package.json:65]: Do we need to raise these dependencies? If yes please align them in the Vue 3 packages. If possible any dependency within the monorepo should be required in the same version.

File: packages/vue2/vue2-vanilla/config/vue.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
-export { computed, defineComponent, inject, onBeforeMount, onUnmounted, reactive, ref, watch, watchEffect } from "@vue/composition-api";
-export type { Ref } from "@vue/composition-api"; 
+export { computed, defineComponent, inject, onBeforeMount, onUnmounted, reactive, ref, watch, watchEffect } from "vue";
+export type { Ref } from "vue"; 
 /**
  * Compatibility type as defineComponent of '@vue/composition-api' can't properly handle PropTypes.
```
Comment (sdirix) [inline packages/vue2/vue2-vanilla/config/vue.ts:7]: also the docs here.

**Review by sdirix — COMMENTED** [review]
Looks good! Did not test it yet, just some questions and remarks.

Thanks for the contribution! :heart:

File: packages/vue2/vue2-vanilla/package.json (hunk 3 above)
Comment (marcianos) [reply packages/vue2/vue2-vanilla/package.json:65]: Yes it should be at least `~4.5.18` for v4 or `~5.0.6 ` for v5 . See [here](https://github.com/vuejs/vue/blob/main/CHANGELOG.md#vue-cli--webpack). I align the Vue 3 packages as well to `~4.5.18`

**Review by sdirix — APPROVED** [review]

**Pull request merged**